sympy/matrices/matrices.py

```diff
@@ -1549,8 +1549,19 @@ def nullity_chain(val):
             i = 2
             while nullity != ret[-1]:
                 ret.append(nullity)
+                if nullity == algebraic_multiplicity:
+                    break
                 nullity = cols - eig_mat(val, i).rank()
                 i += 1
+
+                # XXX Added runtime check. Must improve rank computation
+                # for matrices with complex numbers or symbols.
```

Comment:
Let's change the comment slightly. Remove the `XXX`. Maybe something like, 

```
# Due to issues like #7146 and #15872, SymPy sometimes gives the
# wrong rank. In this case, raise an error instead of returning 
# an incorrect matrix
```

Reply:
`XXX` or `FIXME` can be recognized by some code linters, though 
http://pylint.pycqa.org/en/1.9/technical_reference/features.html?#id19

Reply:
Is `XXX` or `FIXME` used elsewhere in the sympy codebase?

Reply:
Yes
```
$ git grep 'XXX' '*.py'
sympy/assumptions/handlers/sets.py:        # XXX it should be enough to do
sympy/assumptions/satask.py:        # XXX: We need this since True/False are not Basic
sympy/assumptions/sathandlers.py:# XXX: Better name?
sympy/concrete/summations.py:        # XXX remove this test for free_symbols when the default _eval_derivative is in
sympy/core/basic.py:        # XXX: remove this when issue 5169 is fixed
sympy/core/basic.py:            if isinstance(expr_to_call, Symbol):  # XXX When you call a Symbol it is
sympy/core/basic.py:        if kwargs.pop('simultaneous', False):  # XXX should this be the default for dict subs?
sympy/core/basic.py:            # XXX remove the exact flag and make multi-symbol
sympy/core/containers.py:    # XXX: Basic defines count() as something different, so we can't
sympy/core/containers.py:        # XXX: One would expect:
```
```
$ git grep 'FIXME' '*.py'
sympy/core/core.py:# FIXME this is *so* irrelevant and outdated!
sympy/core/facts.py:        # (?) FIXME this is only correct when b & c != null !
sympy/core/tests/test_relational.py:    # FIXME: could replace with random selection after test passes
sympy/core/tests/test_relational.py:    # FIXME: could replace with random selection after test passes
sympy/external/tests/test_codegen.py:            "FIXME: filename extension unknown for language: %s" % language)
sympy/functions/elementary/piecewise.py:        # FIXME: Currently complex intervals are not supported.  A possible
```

Reply:
Interesting. In any case, I think `XXX` should be used to mark code that will be removed from sympy in the near future. I don't think we'll ever be removing this check.